### PR TITLE
Add minimal nn.functional.log_softmax support for NestedTensor

### PR DESCRIPTION
**Pull request by alanhdu**
This only works for the jagged layout and for the non-batch and non-jagged dimensions.

I did this mostly by copy-pasting from the existing softmax implementation, but it seems fairly straightforward and I think it should work.

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]
Looks fine AFAICT, thanks for the contribution :)
